src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbolBase.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.Cci;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a compiler generated and embedded attribute type.
+    /// This type has the following properties:
+    /// 1) It is non-generic, sealed, internal, non-static class.
+    /// 2) It derives from System.Attribute
+    /// 3) It has Microsoft.CodeAnalysis.EmbeddedAttribute
+    /// 4) It has System.Runtime.CompilerServices.CompilerGeneratedAttribute
+    /// 5) It has a parameter-less constructor
+    /// </summary>
+    internal abstract class SynthesizedEmbeddedAttributeSymbolBase : NamedTypeSymbol
```

Comment:
>SynthesizedEmbeddedAttributeSymbolBase [](start = 28, length = 38)

Consider leaving this type in `SynthesizedEmbeddedAttributeSymbol.cs`, along with the derived type, so the changes in this PR are clear.

Reply:
Done.